src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -78,12 +83,23 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
     }
   }
 
-  private def extractFee(tx: ErgoTransaction): Long = {
-    val propositionBytes = settings.chainSettings.monetary.feePropositionBytes
+  private def extractFee(tx: ErgoTransaction): Long =
     ErgoState.boxChanges(Seq(tx))._2
       .filter(_.ergoTree == settings.chainSettings.monetary.feeProposition)
       .map(_.value)
       .sum
+
+  private def randIndexes(qty: Int, max: Int): Seq[Int] = {
+    require(qty <= max)
+    @scala.annotation.tailrec
+    def loop(acc: Seq[Int], qtyLeft: Int): Seq[Int] =
+      if (qtyLeft > 0) {
+        val idx = Random.nextInt(max)
```

Comment:
This is problematic in case qty is close to max. You will have undeterministic number of cycles until you find all indexes.
Your original also was better actually.
The best I can think of is to:
1) put `0 .. max` range into debox.Set.
2) generate random index i in `0 .. set.size`
3) remove using `val wasRemoved = set.remove(i)`
4) check if wasRemoved otherwise try i+1 
This algorithm in O(qty)

Reply:
@oskin1 What is the expected size of max? For quantity it is 200, right?

Reply:
@aslesarenko, there is a requirement require(qty <= max), so loop always terminates. In case qty == max all the indexes in range 0 .. qty are picked.

Reply:
@aslesarenko, upper bound is defined by `ergo.node.mempoolCapacity` and the default is `1000`.